x-pack/solutions/security/plugins/security_solution/public/onboarding/components/onboarding_body/cards/common/connectors/connector_cards.tsx

```diff
@@ -6,42 +6,45 @@
  */
 
 import React, { useCallback } from 'react';
-import { type AIConnector } from '@kbn/elastic-assistant/impl/connectorland/connector_selector';
-import {
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiPanel,
-  EuiLoadingSpinner,
-  EuiText,
-  EuiBadge,
-  EuiSpacer,
-  EuiCallOut,
-  useEuiTheme,
-} from '@elastic/eui';
-import { css } from '@emotion/css';
-import { useKibana } from '../../../../../../common/lib/kibana';
-import {
-  CreateConnectorPopover,
-  type CreateConnectorPopoverProps,
-} from './create_connector_popover';
-import { ConnectorSetup } from './connector_setup';
+import { EuiFlexGroup, EuiFlexItem, EuiLoadingSpinner, EuiCallOut } from '@elastic/eui';
+import { css } from '@emotion/react';
+import type { AIConnector } from './types';
 import * as i18n from './translations';
 import { MissingPrivilegesDescription } from './missing_privileges';
+import { ConnectorSetup } from './connector_setup';
+import { ConnectorSelectorPanel } from './connector_selector_panel';
+import { useStoredAssistantConnectorId } from '../../../../hooks/use_stored_state';
+import { useOnboardingContext } from '../../../../onboarding_context';
 
-interface ConnectorCardsProps
-  extends CreateConnectorPopoverProps,
-    Omit<ConnectorListProps, 'connectors'> {
+interface ConnectorCardsProps {
+  onNewConnectorSaved: () => void;
+  canCreateConnectors?: boolean;
   connectors?: AIConnector[]; // make connectors optional to handle loading state
+  selectedConnectorId?: string;
+  onConnectorSelected: (connector: AIConnector) => void;
 }
 
 export const ConnectorCards = React.memo<ConnectorCardsProps>(
   ({
     connectors,
-    onConnectorSaved,
+    onNewConnectorSaved,
     canCreateConnectors,
     selectedConnectorId,
-    setSelectedConnectorId,
+    onConnectorSelected,
   }) => {
+    const { spaceId } = useOnboardingContext();
+    const [, setStoredAssistantConnectorId] = useStoredAssistantConnectorId(spaceId);
+
+    const onNewConnectorStoredSave = useCallback(
+      (newConnector: AIConnector) => {
+        onNewConnectorSaved();
+        setStoredAssistantConnectorId(newConnector.id);
+        // default select the new connector created
+        onConnectorSelected(newConnector);
+      },
+      [onConnectorSelected, onNewConnectorSaved, setStoredAssistantConnectorId]
+    );
```

Comment:
This is the generic `connector_card` component in the common directory. We should not assume this is only used in the _assistant_. 
The `setStoredAssistantConnectorId` here updates the "assistant card" even when I add a connector in the siem migrations side. The `onConnectorSelected` call will set the stored connector correctly, I think we can remove this:

```suggestion
    const onNewConnectorStoredSave = useCallback(
      (newConnector: AIConnector) => {
        onNewConnectorSaved();
        // default select the new connector created
        onConnectorSelected(newConnector);
      },
      [onConnectorSelected, onNewConnectorSaved]
    );
```